src/python/pants/goal/run_tracker.py

```diff
@@ -229,7 +231,7 @@ def initialize(self, all_options):
 
     self._all_options = all_options
 
-    return run_id
+    return (run_id, run_uuid)
```

Comment:
One way to avoid mixing up objects of the same type (especially strings) that I find useful sometimes is to use a datatype:
```python
from future.utils import text_type

from pants.util.objects import datatype

class RunIds(datatype([('run_id', text_type), ('run_uuid', text_type)])):
  """Can add documentation in the docstring."""
```

You can then `return RunIds(run_id=run_id, run_uuid=run_uuid)` here (I prefer to use keyword arguments when the fields are the same type, like strings). In `reporting.py`, you can then:
```python
    run_ids = run_tracker.initialize(all_options)
    run_dir = os.path.join(self.get_options().reports_dir, run_ids.run_id)
    # ...
    if trace_id is None:
      trace_id = run_ids.run_uuid
```

I personally find it difficult to the read the difference between `run_id` and `run_uuid`, so having the extra context can be helpful.

Reply:
If we get to more than one call site, I would more strongly agree, but with one call site, this feels 50:50 useful vs overheady